src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversion.cs

```diff
@@ -67,6 +67,18 @@ internal bool IsArrayIndex
             }
         }
 
+        private class DeconstructionUncommonData : UncommonData
+        {
+            internal DeconstructionUncommonData(DeconstructionInfo deconstructionInfoOpt, ImmutableArray<Conversion> nestedConversions)
+                : base(false, false, default(UserDefinedConversionResult), null, nestedConversions)
+            {
+                Debug.Assert(!nestedConversions.IsDefaultOrEmpty);
+                _deconstructionInfo = deconstructionInfoOpt;
+            }
+
+            internal DeconstructionInfo _deconstructionInfo;
```

Comment:
>_deconstructionInfo [](start = 40, length = 19)

readonly? Also, consider using name DeconstructionInfo if we intend to access it from outside.